### PR TITLE
Add binaries for h5 compatibility checks

### PR DESCRIPTION
**Pull request by dirkboye**

### CODE REVIEW

File: src/dectris/neggia/plugin/H5ToXds.cpp [hunk 1]
```diff
@@ -177,20 +177,84 @@ void setYPixelSize(H5DataCache* dataCache) {
...
+                    dataCache->pixelMask = read2D<uint32_t>(pixelMask);
+                    break;
+                }
+                case 8: {
+                    auto pm = read2D<uint64_t>(pixelMask);
+                    copyPixelMask(dataCache->pixelMask.get(), pm.get(), s);
+                    break;
+                }
+            }
+        }
```
Comment (kalcutter) [inline src/dectris/neggia/plugin/H5ToXds.cpp]: Shouldn't these switches have `default` labels that throw?

File: bin/check_h5_plugin.cpp [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                    &error_flag);
+    if (error_flag == 0) {
+        std::cerr << "[ OK ] plugin_get_data for last frame successful\n";
+    } else {
+        std::cerr << "[WARN] plugin_get_data for last frame returned error "
+                  << error_flag << "\n";
+        std::cerr << "       does the last data-file exist in directory of "
+                     "master-file?\n";
+        error_flag = 0;
+    }
```
Comment (kalcutter) [inline bin/check_h5_plugin.cpp]: We should just loop through and check all frames.

File: src/dectris/neggia/plugin/H5ToXds.cpp (hunk 1 above)
Comment (dirkboye) [reply src/dectris/neggia/plugin/H5ToXds.cpp]: yes i'll do that to catch any errors in the hdf5 files.

File: bin/check_h5_plugin.cpp (hunk 2 above)
Comment (dirkboye) [reply bin/check_h5_plugin.cpp]: i thought about that in the beginning but wasn't sure if the user always has ALL the data files available for such a test and how long that would take for really huge datasets.

perhaps add a flag to turn off checking all files? (and by default check all files when returning the error for missing data files also print the instructions on how to set the flag)

Comment (kalcutter) [reply bin/check_h5_plugin.cpp]: why does the user want to check their dataset when they don't event have all the data to check? why wouldn't they just test with a smaller dataset?

Comment (kalcutter) [reply bin/check_h5_plugin.cpp]: Lets just keep it simple and always check all frames.

Comment (dirkboye) [reply bin/check_h5_plugin.cpp]: done.

**Comment by kalcutter** [comment]
LGTM

**Pull request merged**